src/components/App/App.js

```diff
@@ -31,27 +83,175 @@ function App() {
-        onOpenSigningPopup={handleSigningPopupOpen}
-      />
-      <Routes>
-        <Route path='/saved-news' element={<SavedNews/>} />
-        <Route path='/' element={<Main/>} />
-      </Routes>
-      <Footer />
-      <SigningPopup isOpen={isSigningPopupOpen} onClose={closeAllPopups} redirectOnClick={handleSignupPopupOpen}></SigningPopup>
-      <SignupPopup isOpen={isSignupPopupOpen} onClose={closeAllPopups} redirectOnClick={handleSigningPopupOpen}></SignupPopup>
-      <Popup isOpen={isMobileNavOpen} onClose={closeAllPopups} name='navbar' />
-    </div>
+    <CurrentUserContext.Provider value={currentUser}>
+      <div className="page">
+        <Header
+          isLoggedIn={isLoggedIn}
+          isMobileNavOpen={isMobileNavOpen} 
+          onClosePopups={closeAllPopups} 
+          onOpenMobileNav={handleMobilePopupOpen}
+          onOpenSigningPopup={handleSigningPopupOpen}
+          onLogout={handleLogout}
+        />
+        <Routes>
+          <Route path='/saved-news' element={
+            <ProtectedRoute
+              isLoggedIn={isLoggedIn}
+              element={SavedNews}
+              isCardsSectionVisible={true}
+              isLoadingCards={false}
+              savedArticles={savedArticles}
+              onSaveArticle={handleSaveArticle} 
+              onDeleteArticle={handleDeleteArticle}
+            />
+          } />
+          <Route path='/' element={
+            <Main 
+              onSubmitSearch={handleSubmitSearch}
+              onSaveArticle={handleSaveArticle}
+              onDeleteArticle={handleDeleteArticle}
+              isLoggedIn={isLoggedIn}
+              savedArticles={savedArticles}
+              isCardsSectionVisible={isCardsSectionVisible}
+              isLoadingCards={isLoadingCards}
+              isFoundCards={isFoundCards}
+              isErrorSearchOccur={isErrorSearchOccur}
+              articles={receivedArticles}
+              keyword={searchKeyword}
+            />
+          } />
+        </Routes>
+        <Footer />
+        <Login 
+          onSubmit={handleLogin}
+          isOpen={isSigningPopupOpen} 
+          onClose={closeAllPopups} 
+          redirectOnClick={handleSignupPopupOpen} 
+          setEmail={setEmail}
+          setPassword={setPassword}
+          globalError={globalError}
+        />
+        { !isRegisteredSuccessPopupOpen? 
+        <Register 
+          onSubmit={handleRegister}
+          isOpen={isSignupPopupOpen} 
+          onClose={closeAllPopups} 
+          redirectOnClick={handleSigningPopupOpen}
+          setEmail={setEmail}
+          setPassword={setPassword}
+          setName={setName}
+          globalError={globalError}
+        />:
+        <RegisteredSuccessPopup 
+          isOpen={isRegisteredSuccessPopupOpen}
+          redirectOnClick={handleSigningPopupOpen}
+          onClose={closeAllPopups} 
+        />
+        }
+        <Popup isOpen={isMobileNavOpen} onClose={closeAllPopups} name='navbar' />
```

Comment:
**Might be better**



If it’s interesting for you here is a way to make a special wrapper for any `popup` with cross button, `Escape` and `overlay` listeners inside. You should create component `Popup`, store it in `Popup.js`. 

```jsx
import { useEffect } from "react";

const Popup = ({ isOpen, name, onClose, children }) => {
// here is `useEffect` for the `Escape` listener
  useEffect(() => {
    // with this we prevent adding the listener if the popup is not opened
    if (!isOpen) return;
// we should define the handler inside `useEffect`, so that it wouldn’t lose the reference to be able to remove it
    const closeByEscape = (e) => {
      if (e.key === 'Escape') {
        onClose();
      }
    }

    document.addEventListener('keydown', closeByEscape)
    // don’t forget to remove the listener in the `clean-up` function
    return () => document.removeEventListener('keydown', closeByEscape)
// here we watch `isOpen` to add the listener only when it’s opened
}, [isOpen, onClose])

// here is the overlay handler
  const handleOverlay = (e) => {
    if (e.target === e.currentTarget) {
        onClose();
    }
  }

// then we add the main wrapper with class `popup` and `popup_opened`
  return (
    <div
      className={`popup ${isOpen ? "popup_opened" : ""} popup_type_${name}`}
      onClick={handleOverlay}
    >
    {/* the container for the contents */}
      <div className='popup__container'>
        {/* here will be anything you add as `children`
*/}
        {children}
        {/* add the close button */}
        <button
          className='popup__close'
          type='button'
          onClick={onClose}
        />
      </div>
    </div>
  );
};

export default Popup;


```

And now you can use `Popup` for any popup in the project:  ` ImagePopup`, ` InfoToolTip` and `PopupWithForm`.

```jsx


function PopupWithForm({isOpen, name, onClose, ...props}) {
  return (
    <Popup isOpen={isOpen} name={name} onClose={onClose}>
        <h2 className='popup__title'>{props.title}</h2>


```

You can make also `Input` for inputs and `Form` for any form in the project.